modin/pandas/test/test_series.py

```diff
@@ -3652,29 +3652,17 @@ def test_str_contains(data, pat, case, na):
 @pytest.mark.parametrize("n", int_arg_values, ids=int_arg_keys)
 @pytest.mark.parametrize("case", bool_arg_values, ids=bool_arg_keys)
 def test_str_replace(data, pat, repl, n, case):
-    modin_series, pandas_series = create_test_series(data)
-
-    try:
-        pandas_result = pandas_series.str.replace(
-            pat, repl, n=n, case=case, regex=False
-        )
-    except Exception as e:
-        with pytest.raises(type(e)):
-            modin_series.str.replace(pat, repl, n=n, case=case, regex=False)
-    else:
-        modin_result = modin_series.str.replace(pat, repl, n=n, case=case, regex=False)
-        df_equals(modin_result, pandas_result)
-
+    eval_general(
```

Comment:
the same test, but wrapped in `eval_general`, that has more proper logic of checking arisen exceptions for equality